packages/dev/core/src/XR/features/WebXRDepthSensing.ts

```diff
@@ -0,0 +1,320 @@
+import { Engine } from "../../Engines/engine";
+import { Observable } from "../../Misc/observable";
+
+/**
+ * Options for Depth Sensing feature
+ */
+export interface IWebXRDepthSensingOptions {
+    /**
+     *  The desired depth sensing usage for the session
+     */
+    usagePreference: XRDepthUsage[];
+    /**
+     * The desired depth sensing data format for the session
+     */
+    dataFormatPreference: XRDepthDataFormat[];
+}
+
+type GetDepthInMetersType = (x: number, y: number) => number;
+
+/**
+ * WebXR Feature for WebXR Depth Sensing Module
+ */
+export class WebXRDepthSensing extends WebXRAbstractFeature {
+    private _cachedDepthInfo?: XRDepthInformation;
+
+    /**
+     * Width of depth data. If depth data is not exist, returns null.
+     */
+    public get width(): number | null {
+        if (!this._cachedDepthInfo) {
+            return null;
+        }
+        return this._cachedDepthInfo.width;
+    }
+
+    /**
+     * Height of depth data. If depth data is not exist, returns null.
+     */
+    public get height(): number | null {
+        if (!this._cachedDepthInfo) {
+            return null;
+        }
+        return this._cachedDepthInfo.height;
+    }
+
+    /**
+     * Scale factor by which the raw depth values must be multiplied in order to get the depths in meters.
+     */
+    public get rawValueToMeters(): number | null {
+        if (!this._cachedDepthInfo) {
+            return null;
+        }
+
+        return this._cachedDepthInfo.rawValueToMeters;
+    }
+
+    /**
+     * An XRRigidTransform that needs to be applied when indexing into the depth buffer.
+     */
+    public get normDepthBufferFromNormView(): XRRigidTransform | null {
+        if (!this._cachedDepthInfo) {
+            return null;
+        }
+
+        return this._cachedDepthInfo.normDepthBufferFromNormView;
+    }
+
+    /**
+     * Describes which depth-sensing usage ("cpu-optimized" or "gpu-optimized" is used.
+     *
+     */
+    public get depthUsage(): XRDepthUsage {
+        return this._xrSessionManager.session.depthUsage;
+    }
+
+    /**
+     * Describes which depth sensing data format ("luminance-alpha" or "float32") is used.
+     */
+    public get depthDataFormat(): XRDepthDataFormat {
+        return this._xrSessionManager.session.depthDataFormat;
+    }
+
+    /**
+     * Latest cached WebGLTexture which containing depth buffer information.
+     * This can be used when the depth usage is gpu-optimized.
+     */
+    public get latestWebGLTexture(): WebGLTexture | null {
+        if (!this._cachedDepthInfo) {
+            return null;
+        }
+
+        const webglDepthInfo = this._cachedDepthInfo as XRWebGLDepthInformation;
+        if (!webglDepthInfo.texture) {
+            return null;
+        }
+
+        return webglDepthInfo.texture;
+    }
+
+    /**
+     * cached depth buffer
+     */
+    public get latestDepthBuffer(): ArrayBufferView | null {

```

Comment:
nit:
```suggestion
    public get latestDepthBuffer(): Nullable<ArrayBufferView> {
```

Reply:
Thank you. I'd like to apply your suggestion